internal/blobtesting/versionedmap.go

```diff
@@ -0,0 +1,248 @@
+package blobtesting
+
+import (
+	"bytes"
+	"context"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/repo/blob"
+)
+
+type entry struct {
```

Comment:
It seems that the intent is to implement not only versioning, but object locking as well. It'd make sense to rename the data structs, functions and files to reflect that.
=> `objectLockingMap` or similar.

Reply:
Will do.